snapcraft/internal/meta.py

```diff
@@ -137,7 +139,17 @@ def _record_snapcraft(self):
         if os.environ.get('SNAPCRAFT_BUILD_INFO'):
             os.makedirs(record_dir, exist_ok=True)
             with open(record_file_path, 'w') as record_file:
-                yaml.dump(self._config_data, record_file)
+                annotated_snapcraft = self._annotate_snapcraft(
+                    copy.deepcopy(self._config_data))
+                yaml.dump(annotated_snapcraft, record_file)
+
+    def _annotate_snapcraft(self, data):
+        for part in data['parts']:
+            pull_state = get_state(
```

Comment:
wouldn't this be better handled in the `state` package? A `get_state(part_name, state)` doing the right thing so that

```python
pull_stage = get_state(part_name, state)
```

IOW, `get_state` shouldn't need exposing paths for the case where these change.